File: Python/Valid Sudoku.py
class Solution:
    # @param board, a 9x9 2D array
    # @return a boolean
    def isValidSudoku(self, board):
		if len(board) == 0:
			return False
		for i in range(0,9):
			row = [0 for j in xrange(9)]
			col = [0 for j in xrange(9)]
			for j in range(0,9):
				if board[i][j] != ".":
					if row[int(board[i][j])-10] == 1:
						return False
					row[int(board[i][j])-10] = row[int(board[i][j])-10] + 1
				if board[j][i] != ".":
					if col[int(board[j][i])-10] == 1:
						return False
					col[int(board[j][i])-10] = col[int(board[j][i])-10] + 1
		for i in range(0,9,3):
			for j in range(0,9,3):
				row = [0 for u in xrange(9)]
				for t in range(0,3):
					for k in range(0,3):
						if board[t+i][k+j] == ".":
							continue
						if row[int(board[t+i][k+j])-10] == 1:
							return False
						row[int(board[t+i][k+j])-10] = row[int(board[t+i][k+j])-10] + 1
		return True

File: Java/Valid Sudoku.java
class Valid SudokuP{
	public boolean isValidSudoku(char[][] board) {
        //check row and col
		if (board.length == 0)
            return false;
        for(int i=0;i<board.length;i++){
        	int[] row = new int[9];
        	int[] col = new int[9];
        	for(int j=0;j<board[0].length;j++){
        		if(board[i][j] != '.'){
        			int rowIndex = board[i][j] - '1';
        			if(row[rowIndex] == 0)
        				row[rowIndex] = 1;
        			else
        				return false;
        		}
        		if(board[j][i] != '.'){
        			int colIndex = board[j][i] - '1';
        			if(col[colIndex] == 0)
        				col[colIndex] = 1;
        			else
        				return false;
        		}
        	}
        	
        }
        //check 3*3 board
        for(int i=0;i<board.length;i+=3){
        		for(int j=0;j<board[0].length;j+=3){
        			int[] block = new int[9];
        			for(int t=0;t<3;t++){
        				for(int k=0;k<3;k++){
        					if(board[i+t][j+k] != '.'){
        						int index = (board[i+t][j+k]-'1');
        						if(block[index] == 0)
        							block[index] = 1;
        						else
        							return false;
        					}
        				}
        			}
        		}
        	}
        	return true;
    }
}